src/App.jsx

```diff
@@ -4,18 +4,21 @@ import Header from './components/Header'
 import Homepage from './components/Homepage'
 import { Route, Routes } from 'react-router-dom'
 import ArticleProvider from './components/ArticleProvider'
+import SingleArticle from './components/SingleArticle.jsx'
 
 function App() {
 
-  const [article, setCurrentArticle] = useState({});
+const [isLoading, setIsLoading] = useState(true);
```

Comment:
I would give components their own isLoading rather than sharing. But if it works, no problem 🙂